internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -405,6 +405,30 @@ func TestMachineSetReconciler(t *testing.T) {
 
 		// Validate that the controller set the cluster name label in selector.
 		g.Expect(instance.Status.Selector).To(ContainSubstring(testCluster.Name))
+
+		t.Log("Verifying MachineSet can be scaled down when templates don't exist")
+		g.Expect(env.Delete(ctx, bootstrapTmpl)).To(Succeed())
+		g.Expect(env.Delete(ctx, infraTmpl)).To(Succeed())
```

Comment:
```suggestion
		g.Expect(env.CleanupAndWait(ctx, bootstrapTmpl)).To(Succeed())
		g.Expect(env.CleanupAndWait(ctx, infraTmpl)).To(Succeed())
```
Just to make sure the objects are also gone from the cache